### ISSUE TITLE
Network: cannot disable physics on stabilization for some nodes

### ISSUE THREAD

**Issue by rools32** [almende/vis#1782]
I have an event that disables physics on nodes and edges when the graph is stabilized. But, when I drag the center node (d05e), other nodes move. What is wrong in my code?
See the atached files for details.
[data.txt](https://github.com/almende/vis/files/208395/data.txt)
[network.html](https://github.com/almende/vis/files/208394/network.txt)

**Comment by AlexDM0** [comment]
since your network.html is corrupt i can only guess. If i were to guess: you're not using the dataset to update the node.

**Comment by rools32** [comment]
Hum, I can open my network.html without any problem... Strange...
This is what I use to disable physics:

    network.on("stabilizationIterationsDone", function () {
        network.setOptions({
            nodes: {physics: false},
            edges: {physics: false},
        });
    });

**Comment by AlexDM0** [comment]
if I try the setOptions in our examples using the console it works. 
http://visjs.org/examples/network/basicUsage.html

So.. is your event firing?

**Comment by rools32** [comment]
Yes it is fired. And the disabling works most of the time, but not with the data set I provided. If I remove one node from it, it will probably work.

**Comment by AlexDM0** [comment]
I'm not sure how you figure I can help you here..

**Comment by rools32** [comment]
This is my html code:

    <!doctype html>
    <html>
    <head>
        <script type="text/javascript" src="visdist/vis.js"></script>
        <link href="visdist/vis.css" rel="stylesheet" type="text/css"/>
        <input type="file" id="files" name="files[]"/>
    
    </head>
    <body>
    
        <div id="mynetwork"></div>
    
    <script type="text/javascript">
    
    function handleFile(evt) {
        var files = evt.target.files;
        var fr = new FileReader();
        fr.onload = function(e) {
            data = e.target.result;
            loadFile(data);
        };
        fr.readAsText(files[0]);
    }
    
    document.getElementById('files').addEventListener('change', handleFile, false);
    
    function loadFile(data)
    {
        var mydata = JSON.parse(data);
        var nodes = new vis.DataSet({});
        var edges = new vis.DataSet({});
    
        edges.add(mydata.edges);
        nodes.add(mydata.nodes);
    
        nodes.forEach(function(n) {
            nodes.update({id: n.id, title:n.id});
        });
    
        // create a network
        var container = document.getElementById('mynetwork');
        var data = {
            nodes: nodes,
            edges: edges
        };
    
        var options = {interaction:{hover:true, tooltipDelay:0, multiselect: true},
            physics: true,
        };
        var network = new vis.Network(container, data, options);
    
        network.on("stabilizationIterationsDone", function () {
            network.setOptions({
                nodes: {physics: false},
                edges: {physics: false},
            });
        });
    }
    
    </script>
    
    </body>
    </html>

**Comment by AlexDM0** [comment]
So it doesn't work with the dataset you provide, which is the only thing thats not in your code above?

**Comment by rools32** [comment]
That's right.

**Comment by AlexDM0** [comment]
I see, you can turn off improvedLayout in the meantime.

We'll see if we can get it fixed.

**Comment by rools32** [comment]
Thanks for the hack. Unfortunately, my network has more than 7,000 nodes so I am not sure this solution will be possible.

**Comment by AlexDM0** [comment]
It's not a hack. You can also set physics:false globally. Or set physics false per node and edge. There in a bookkeeping issue in the clustering algorithm used by de improvedLayout algorithm. Your call

**Comment by rools32** [comment]
I used hack, to say it handles my problem, but I cannot use it because it is too slow: I tested it with 100 nodes the rendering time was 12 seconds instead of 6. My complete graph has 7000 nodes and is rendered in 50 minutes with improvedLayout on.
The code I gave was just a reproducer, but my code is a lot more complicated. I need physcis because I want to move some nodes and keep physic constraints. I already set physics per node and edge to enable it for some and disable for others when I move them, but the bug is still present.
